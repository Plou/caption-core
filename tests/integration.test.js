const CaptionCore = require("./../dist/index");

describe("CaptionCore", () => {
  describe("textSearch", () => {
    it("retrieves fastests results", done => {
      CaptionCore.searchByQuery("Mr Robot S01E02", "eng").on(
        "fastest",
        results => {
          expect(results).toMatchSnapshot();
          done();
        },
      );
    });
    it("retrieves all results", done => {
      CaptionCore.searchByQuery("Mr Robot S01E02", "eng").on(
        "completed",
        results => {
          expect(results).toMatchSnapshot();
          done();
        },
      );
    });
  });
});
