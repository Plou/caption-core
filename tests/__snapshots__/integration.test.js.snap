// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptionCore textSearch retrieves all results 1`] = `
Array [
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19dc0c5c/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1954809951",
    "extention": "",
    "name": "Mr.Robot.S01E02.HDTV.x264-KILLERS.srt",
    "score": 4,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19de0c5e/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1954809953",
    "extention": "",
    "name": "Mr.Robot.S01E02.HDTV.x264-KILLERS.srt",
    "score": 4,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19ba0c58/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1955334557",
    "extention": "",
    "name": "Mr._Robot.S01E02.EN.srt",
    "score": 0,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19b20c4f/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1954801901",
    "extention": "",
    "name": "Mr.Robot.S01E02.HDTV.x264-KILLERS.srt",
    "score": 0,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19b80c56/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1955045952",
    "extention": "",
    "name": "Mr.Robot.1E02.720p.BluRay.x264-DEMAND.srt",
    "score": 0,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19c20c58/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1954801892",
    "extention": "",
    "name": "English.srt",
    "score": 0,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19d10c5e/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1954813886",
    "extention": "",
    "name": "Mr Robot S01E02 x264.srt",
    "score": 0,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": "http://dl.opensubtitles.org/en/download/src-api/vrf-19da0c63/sid-ko35cHkxryjuh-AAXy33-dzMi1d/filead/1955069887",
    "extention": "",
    "name": "Mr Robot S01 E02 720p Bluray x264- CPUL.srt",
    "score": 0,
    "size": "",
    "source": "opensubtitles",
  },
  Object {
    "download": Object {
      "distribution": "UNKNOWN",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/1",
      "referer": "/show/5151",
      "team": "KILLERS",
      "version": "KILLERS",
    },
    "extention": "",
    "name": "Mr Robot S01e02.UNKNOWN.KILLERS",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/7",
      "referer": "/show/5151",
      "team": "NTB",
      "version": "WEB-DL-NTB",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/18",
      "referer": "/show/5151",
      "team": "NTB(UNCENSORED)",
      "version": "WEB-DL-NTB(UNCENSORED)",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB(UNCENSORED)",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "UNKNOWN",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/0",
      "referer": "/show/5151",
      "team": "KILLERS",
      "version": "KILLERS",
    },
    "extention": "",
    "name": "Mr Robot S01e02.UNKNOWN.KILLERS",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/8",
      "referer": "/show/5151",
      "team": "NTB",
      "version": "WEB-DL-NTB",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/19",
      "referer": "/show/5151",
      "team": "NTB(UNCENSORED)",
      "version": "WEB-DL-NTB(UNCENSORED)",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB(UNCENSORED)",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "UNKNOWN",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/20",
      "referer": "/show/5151",
      "team": "SHAANIG",
      "version": "SHAANIG",
    },
    "extention": "",
    "name": "Mr Robot S01e02.UNKNOWN.SHAANIG",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "BLURAY",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/15",
      "referer": "/show/5151",
      "team": "DEMAND",
      "version": "BDRIP.X264-DEMAND",
    },
    "extention": "",
    "name": "Mr Robot S01e02.BLURAY.DEMAND",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "BLURAY",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/16",
      "referer": "/show/5151",
      "team": "DEMAND",
      "version": "BDRIP.X264-DEMAND",
    },
    "extention": "",
    "name": "Mr Robot S01e02.BLURAY.DEMAND",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
]
`;

exports[`CaptionCore textSearch retrieves fastests results 1`] = `
Array [
  Object {
    "download": Object {
      "distribution": "UNKNOWN",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/1",
      "referer": "/show/5151",
      "team": "KILLERS",
      "version": "KILLERS",
    },
    "extention": "",
    "name": "Mr Robot S01e02.UNKNOWN.KILLERS",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/7",
      "referer": "/show/5151",
      "team": "NTB",
      "version": "WEB-DL-NTB",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/18",
      "referer": "/show/5151",
      "team": "NTB(UNCENSORED)",
      "version": "WEB-DL-NTB(UNCENSORED)",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB(UNCENSORED)",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "UNKNOWN",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/0",
      "referer": "/show/5151",
      "team": "KILLERS",
      "version": "KILLERS",
    },
    "extention": "",
    "name": "Mr Robot S01e02.UNKNOWN.KILLERS",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/8",
      "referer": "/show/5151",
      "team": "NTB",
      "version": "WEB-DL-NTB",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "WEB-DL",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/19",
      "referer": "/show/5151",
      "team": "NTB(UNCENSORED)",
      "version": "WEB-DL-NTB(UNCENSORED)",
    },
    "extention": "",
    "name": "Mr Robot S01e02.WEB-DL.NTB(UNCENSORED)",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "UNKNOWN",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/20",
      "referer": "/show/5151",
      "team": "SHAANIG",
      "version": "SHAANIG",
    },
    "extention": "",
    "name": "Mr Robot S01e02.UNKNOWN.SHAANIG",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "BLURAY",
      "lang": "English",
      "langId": "eng",
      "link": "/updated/1/101990/15",
      "referer": "/show/5151",
      "team": "DEMAND",
      "version": "BDRIP.X264-DEMAND",
    },
    "extention": "",
    "name": "Mr Robot S01e02.BLURAY.DEMAND",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
  Object {
    "download": Object {
      "distribution": "BLURAY",
      "lang": "English",
      "langId": "eng",
      "link": "/original/101990/16",
      "referer": "/show/5151",
      "team": "DEMAND",
      "version": "BDRIP.X264-DEMAND",
    },
    "extention": "",
    "name": "Mr Robot S01e02.BLURAY.DEMAND",
    "score": 0,
    "size": "",
    "source": "addic7ed",
  },
]
`;
